src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java

```diff
@@ -133,7 +133,7 @@
   private final ProjectBindingManager bindingManager;
   private final EventWatcher watcher;
   private final LanguageClientLogOutput lsLogOutput;
-  private final Map<String, Boolean> filesIgnoredByScmCache = new HashMap<>();
+  private final ScmIgnoredCache filesIgnoredByScmCache;
```

Comment:
Nothing to do with this commit, but I believe that this cache can become stale when the `.gitignore` is modified (e.g a file that was not ignored can become ignored, and the cache won't know it until the user closes/reopen the file).

Reply:
Yes, this is what I suggested to Kirill to be an acceptable limitation.